### DISCUSSION TITLE
Modified Plot doesn't work with Plotly

### DISCUSSION THREAD

**Discussion by RanggaGemilang** [teamalgoritma/community#227] [Q&A] [answered]
Hi hi, 

1. So I've been trying to make my "modified plot" interactive using **ggplotly**, however the plot is not visible in the result. Is there any way to fix this or is there any other interactive library that can accomodate my plot? 
2. How to fix this cropped plot? _notice the red circle at bottom-right_

Here's the static plot:
![question1](https://user-images.githubusercontent.com/107049922/181183314-e12c313d-f5f2-4d9c-90b2-bc5df0238641.png)

Here's after **ggplotly** applied:
![question2](https://user-images.githubusercontent.com/107049922/181183458-d43c6ad7-f876-4113-8488-91d78abf93ba.png)

_Note: I'm using **ggfx** library to modify this plot._

Thank you!

**Comment by idoali** [comment, marked as answer]
Selamat malam pak

Untuk masalah pertama, bapak bisa menambahkan fungsi `xlim()`. Dalam kasus bapak, maka akan menjadi `xlim(1965, 2017)`, karena rentang tahunnya antara 1965 sampai 2017. Contohnya adalah seperti ini 

```{R}
ggplot(data, aes(x=year, y=sales)) +
      geom_point() +
      geom_line() + 
      xlim(1965, 2017)
```

Untuk masalah kedua, coba bapak membuatnya dengan cara seperti ini dulu :
```{R}
p <- ggplot(data, aes(x=year, y=sales))
p <- p + geom_point() + geom_line() + xlim(1965, 2017)
ggplotly(p)
```
Jadi kita membuat object `p` yang hanya memiliki ggplot canvas nya saja. Kemudian object `p` tadi kita tambahkan dengan fungsi-fungsi ggplot yang lain. Baru setelah itu, object `p` kita masukkan ke dalam fungsi `ggplotly()`.

Semoga jawabannya bisa membantu pak.
